llvm/test/Feature/Repo/repo_pruning_GV_with_zeroinitializer.ll

```diff
@@ -0,0 +1,36 @@
+; This testcase tests that hash generation for GV with zeroinitializer.
+;
+; If a global variable has zeroinitializer during the RepoMetadataGenerationPass and
+; its initial value changes after the RepoMetadataGenerationPass, the digest of GV
+; needs to be recalculated.
+
+; The testcase includes three steps:
+; Step 1: Build the test Inputs/repo_GV_with_zeroinitializer.ll and create the database 'clang.db' which contains the Tickets.
+; Step 2: Build this file and check the function '-b' doesn't be pruned.
+
+; This test only works for the Debug build because the digest of A is calculated for the Debug build.
+
+; RUN: rm -f %t.db
+; RUN: env REPOFILE=%t.db opt -O3 -S -mtriple=x86_64-pc-linux-gnu-repo %S/Inputs/repo_GV_with_zeroinitializer.ll  -o %t.ll
+; RUN: env REPOFILE=%t.db llc -mtriple="x86_64-pc-linux-gnu-repo" -filetype=obj %t.ll -o %t.o
+; RUN: env REPOFILE=%t.db opt -O3 -S -mtriple=x86_64-pc-linux-gnu-repo %s | FileCheck %s
+
+target triple = "x86_64-pc-linux-gnu-repo"
```

Comment:
This code looks like a copy of Inputs/repo_GV_with_zeroinitializer.ll. What's the reason for having both?

Reply:
These two tests are used to reproduce the issue mentioned in issue25, the comments https://github.com/SNSystems/llvm-project-prepo/issues/25#issuecomment-492531238 and https://github.com/SNSystems/llvm-project-prepo/issues/25#issuecomment-492532296 give the detailed reason. Please let me know if it is not clear.

Reply:
Two tests? I see two files (`Repo/Inputs/repo_GV_with_zeroinitializer.ll` and `Repo/repo_pruning_GV_with_zeroinitializer.ll `) but only a single test (in the latter file). Is there any difference in the code in the two files?

Reply:
> 
> 
> Two tests? I see two files (`Repo/Inputs/repo_GV_with_zeroinitializer.ll` and `Repo/repo_pruning_GV_with_zeroinitializer.ll `) but only a single test (in the latter file). 
>
Sorry, Yes, you are right. Only one test.

>Is there any difference in the code in the two files?

The Repo/Inputs/repo_GV_with_zeroinitializer.ll contains variables .str.a  and _a, both have a private/internal linkage type.

The Repo/repo_pruning_GV_with_zeroinitializer.ll contains variables .str.b and _b, both have a private/internal linkage type.

As described in the issue 25, '_a' has a xfixup to symbol '.str.a', which exists in the Repo/Inputs/repo_GV_with_zeroinitializer.ll and has a internal linkage type. The '.str.a' is not visible to the Repo/repo_pruning_GV_with_zeroinitializer.ll.

The '_b' has a xfixup to symbol '.str.b', which exists in the Repo/repo_pruning_GV_with_zeroinitializer.ll and has a internal linkage type. The '.str.b' is not visible to the Repo/Inputs/repo_GV_with_zeroinitializer.ll. Therefore, the '_a' and '_b' should have the different digest values.

The Repo/repo_pruning_GV_with_zeroinitializer.ll tests whether the variables '_a' and '_b' have the different digest values (not share the same fragment) by checking the pruning status in the TicketNode matadata.

Before the fix, the pruning status is true since '_a' and '_b' have the same digest value. The test failed. After the fix, the pruning status is false since the digests of '_a' and '_b' have been recalculated during the globalopt pass and they are different.

Reply:
> > Is there any difference in the code in the two files?
> The Repo/Inputs/repo_GV_with_zeroinitializer.ll contains variables .str.a and _a, both have a private/internal linkage type.
> The Repo/repo_pruning_GV_with_zeroinitializer.ll contains variables .str.b and _b, both have a private/internal linkage type.

They're the same apart from the variable names?

Reply:
The string values of .str.a and .str.b are different as well.

@.str.a = private unnamed_addr constant [6 x i8] c"**Hello\00**", align 1
@.str.b = private unnamed_addr constant [7 x i8] c"**World!\00**", align 1

Reply:
Is there a way to simplify the test code or to make the differences easier to spot?